### PR TITLE
cpu/nrf5x: implement periph_timer_periodic

### PR DESCRIPTION
**Pull request by chrysn**
### Contribution description

This adds the periph_timer_periodic feature to nrf5x boards.

I have little experience with the nrf5x hardware registers, but the documentation seems to be quite straightforward.

### Testing procedure

With any nrf5x board, run

```
$ make -C tests/periph_timer_periodic all flash test
```

Quite weirdly, it fails for the nrf52840dongle I'm developing this on by hanging; adding

```
    for (volatile int i = 0; i < 4000000; ++i) {
    }
```

to `tests/periph_timer_periodic/main.c` in main before mutex_lock makes it pass up to the point where the added time messes things up.

It does work with a particle-xenon however, leading me to assume that there may be some interference between this test and the USB stdio; any experimentation by someone who has a board with pre-existing periph_timer_periodic and USB stdio would be appreciated.

### Issues/PRs references

Inspiration for the implementation was drawn from https://github.com/RIOT-OS/RIOT/pull/13902 (although no inline functions were used to set/clear the periodic flag to stay consistent with the "channel active" flags present right next there).

This is primarily added to allow testing https://github.com/RIOT-OS/RIOT/pull/13908 (but getting a DAC in here will be harder).

[edit: added 13908 reference]

### CODE REVIEW

File: cpu/nrf5x_common/periph/timer.c [hunk 1]
```diff
@@ -106,6 +107,30 @@ int timer_set_absolute(tim_t tim, int chan, unsigned int value)
...
+{
+    /* see if channel is valid */
+    if (chan >= timer_config[tim].channels) {
+        return -1;
+    }
+
+    ctx[tim].flags |= (1 << chan);
+    ctx[tim].is_periodic |= (1 << chan);
+    dev(tim)->CC[chan] = value;
+    (void)flags;
```
Comment (benpicco) [inline cpu/nrf5x_common/periph/timer.c:120]:
```suggestion
```

it's uses right on the next line

**Review by benpicco — COMMENTED** [review]
looks good - Kconfig will need an update too

**Comment by benpicco** [comment]
I can confirm that this works well on `nrf52840dk`, no regressions with `tests/periph_timer`

File: cpu/nrf5x_common/periph/timer.c (hunk 1 above)
Comment (chrysn) [reply cpu/nrf5x_common/periph/timer.c]: Thanks, that was a leftover from experimentation.

**Comment by benpicco** [comment]
Please squash!

**Review by benpicco — APPROVED** [review]

**Comment by chrysn** [comment]
Kconfig added -- I'm not sure how to test that, though, but it seems straightforward enough.

I've just given this additional testing by combining it with #13908, and it produced intelligible voice output.

**Comment by benpicco** [comment]
> Kconfig added -- I'm not sure how to test that, though

`tests/kconfig_features` takes care of that (and CI will fail if Kconfig features do not match 'classic' Make features)

**Pull request merged**